src/services/fileServices/MdPageServices/PageService.ts

```diff
@@ -116,33 +115,40 @@ export class PageService {
    * @param sessionData session credentials of the user
    */
   parsePageName = (
-    pageName: string,
+    pathInfo: PathInfo,
     sessionData: UserSessionData
   ): ResultAsync<PageName, NotFoundError | MissingResourceRoomError> =>
-    this.parseHomepage(pageName)
+    this.isMarkdownPage(pathInfo.name)
+      .andThen(() => this.parseHomepage(pathInfo))
       // NOTE: Order is important as `contact-us` and unlinked pages
       // are both rooted at `/pages`
-      .orElse(() => this.parseContactUsPage(pageName))
-      .orElse(() => this.parseUnlinkedPages(pageName))
+      .orElse(() => this.parseContactUsPage(pathInfo))
+      .orElse(() => this.parseUnlinkedPages(pathInfo))
       .asyncAndThen<PageName, NotFoundError>(okAsync)
       // NOTE: We read the `_config.yml` to determine if it is a resource page.
       // If it is not, we assume that this is a collection page.
       // Because this method is invoked on existing file paths from the frontend,
       // this assumption will hold true.
-      .orElse(() => this.parseResourceRoomPage(pageName, sessionData))
-      .orElse(() => this.parseCollectionPage(pageName))
+      .orElse(() => this.parseResourceRoomPage(pathInfo, sessionData))
+      .orElse(() => this.parseCollectionPage(pathInfo))
+
+  isMarkdownPage = (pageName: string): Result<string, PageParseError> => {
+    if (pageName.endsWith(".md")) return ok(pageName)
```

Comment:
Nit: can we just return a boolean value instead? Using neverthrow here feels too verbose for a function like this

Reply:
actually the problem here is that we want to distinguish `MarkdownPages` from other things (eg: `YmlConfig` or `HtmlPages`) - the problem with returning `boolean` is that we lose context wrt what the `pageName` is beyond its immediate containing function.

i can change this to `string is MarkdownPage` and brand this string to prevent the verbosity - wdyt

Reply:
hmm isnt that beyond the scope of this function, which is to just tell if this page is of type `MarkdownPages`? 
if not should we then rename this to `getPageType` right?